examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -0,0 +1,211 @@
+# models performance in terms of
+# :func:`~metrics.mean_absolute_percentage_error` and fit times.
+
+from sklearn.model_selection import cross_validate
+import matplotlib.pyplot as plt
+
+scoring = "neg_mean_absolute_percentage_error"
+dropped_result = cross_validate(hist_dropped, X, y, cv=3, scoring=scoring)
+one_hot_result = cross_validate(hist_one_hot, X, y, cv=3, scoring=scoring)
+ordinal_result = cross_validate(hist_ordinal, X, y, cv=3, scoring=scoring)
+native_result = cross_validate(hist_native, X, y, cv=3, scoring=scoring)
+
+
+def plot_results(figure_title):
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 8))
+
+    plot_info = [('fit_time', 'Fit times (s)', ax1, None),
+                 ('test_score', 'Mean Absolute Percentage Error', ax2,
+                  (0, 0.20))]
+
+    x, width = np.arange(4), 0.9
+    for key, title, ax, y_limit in plot_info:
+        items = [dropped_result[key], one_hot_result[key], ordinal_result[key],
+                 native_result[key]]
+        ax.bar(x, [np.mean(np.abs(item)) for item in items],
+               width, yerr=[np.std(item) for item in items],
+               color=['C0', 'C1', 'C2', 'C3'])
+        ax.set(xlabel='Model', title=title, xticks=x,
+               xticklabels=["Dropped", "One Hot", "Ordinal", "Native"],
+               ylim=y_limit)
+    fig.suptitle(figure_title)
+
+
+plot_results("Gradient Boosting on Adult Census")
+
+# %%
+# We see that the model with one-hot-encoded data is by far the slowest. This
+# is to be expected, since one-hot-encoding creates one additional feature per
+# category value (for each categorical feature), and thus more split points
+# need to be considered during fitting. The native handling of categorical
+# features is slightly slower than treating categories as ordered quantities
+# ('Ordinal'), since native handling requires :ref:`sorting categories
+# <categorical_support_gbdt>`.
+#
+# In terms of prediction performance, dropping the categorical features leads
+# to poorer performance. The three models that use categorical features have
+# comparable error rates, with a slight edge for the native handling.
+#
+# In general, one can expect poorer predictions from one-hot-encoded data,
+# especially when the the trees depths or the number of nodes are limited: with
+# one-hot-encoded data, one needs more split points (i.e. more depth) in order
+# to recover an equivalent split that could be obtained in one single split
+# point with native handling. This is also true when categories are treated as
+# ordinal quantities: if categories are `A..F` and the best split is `ACF -
+# BDE` the one-hot-encoder model will need 3 split points (one per category in
+# the left node), and the ordinal non-native model will need 4 splits: 1 split
+# to isolate `A`, 1 split to isolate `F`, and 2 splits to isolate `C` from
+# `BCDE`.
+
+# %%
+# Under-fitting variant
+# ----------------------
```

Comment:
Since this is a very specific kind of overfitting I would suggest to be more precise:

```suggestion
# Limitting the number of splits
# ------------------------------
```